internal/controller/services/secretgenerator/secret_test.go

```diff
@@ -79,12 +79,14 @@ func TestNewSecret(t *testing.T) {
 				assert.Contains(t, err.Error(), tt.expectedErrMessage)
 			case "nil":
 				require.NoError(t, err)
+				assert.Empty(t, secret.Value)
 			case "success":
 				require.NoError(t, err)
 				require.NotNil(t, secret)
 				assert.Equal(t, tt.secretName, secret.Name)
 				assert.Equal(t, tt.secretType, secret.Type)
 				assert.Equal(t, tt.complexity, secret.Complexity)
```

Comment:
You can add also these checks to the nil scenario but otherwise looks good.

Reply:
Added, @MarianMacik please review